feat(dew): add a new one-time resource to sign data (#7727)
huaweicloud/terraform-provider-huaweicloud: docs/resources/kms-sign.md

---
subcategory: "Data Encryption Workshop (DEW)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_kms_sign"
description: |-
  Manages a KMS sign resource within HuaweiCloud.
---

# huaweicloud_kms_sign

Manages a KMS sign resource within HuaweiCloud.

-> The current resource is a one-time resource, and destroying this resource will not recover sign,
but will only remove the resource information from the tfstate file.

## Example Usage

```hcl
variable "key_id" {}
variable "message" {}
variable "signing_algorithm" {}

resource "huaweicloud_kms_sign" "test" {
  key_id               = var.key_id
  message              = var.message
  signing_algorithm    = var.signing_algorithm

> }

Reviewer: ```suggestion
resource "huaweicloud_kms_sign" "test" {
  key_id            = var.key_id
  message           = var.message
  signing_algorithm = var.signing_algorithm
}
```

```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.
  Changing this creates a new resource.

* `key_id` - (Required, String, NonUpdatable) Specifies the key ID.
  The key only supports asymmetric keys with key_usage set to SIGN_VERIFY.
  The valid length is `36` bytes, meeting regular match **^[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}$**.
  For example: **0d0466b0-e727-4d9c-b35d-f84bb474a37f**.

* `message` - (Required, String, NonUpdatable) Specifies the message digest or message to be encrypted.

>   The message length must be less than 4096 bytes and should be encoded using Base64.

Reviewer: ```suggestion
* `message` - (Required, String, NonUpdatable) Specifies the message digest or message to be signed.
  The message length must be less than `4,096` bytes and should be encoded using Base64.
```

* `signing_algorithm` - (Required, String, NonUpdatable) Specifies the signature algorithm.
  The valid values are as follow:
  + **SM2DSA_SM3** : Only supports signing message digests.
  + **RSASSA_PSS_SHA_256**
  + **RSASSA_PSS_SHA_384**
  + **RSASSA_PSS_SHA_512**
  + **RSASSA_PKCS1_V1_5_SHA_256**
  + **RSASSA_PKCS1_V1_5_SHA_384**
  + **RSASSA_PKCS1_V1_5_SHA_512**
  + **ECDSA_SHA_256**
  + **ECDSA_SHA_384**
  + **ECDSA_SHA_512**

* `message_type` - (Optional, String, NonUpdatable) Specifies the type of the message.

>   The valid values are as follow, the default value is **DIGEST**.

Reviewer: ```suggestion
  The valid values are as follows, the default value is **DIGEST**.
```

  + **DIGEST**: Message digest.
  + **RAW**: Original message.

* `sequence` - (Optional, String, NonUpdatable) Specifies the sequence number of the request message, `36` bytes.
  For example: **919c82d4-8046-4722-9094-35c3c6524cff**.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

> * `signature` - The Signature value, encoded using Base64.

Reviewer: ```suggestion
* `signature` - The signature value, encoded using Base64.
```

---
subcategory: "Data Encryption Workshop (DEW)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_kms_sign"
description: |-
  Manages a KMS sign resource within HuaweiCloud.
---

# huaweicloud_kms_sign

Manages a KMS sign resource within HuaweiCloud.

-> 1. The current resource is a one-time resource, and destroying this resource will not recover sign,
but will only remove the resource information from the tfstate file.
<br/>2. Only asymmetric keys with key_usage set to SIGN_VERIFY are supported for signing operations.

> <br/>3. When using SM2 keys for signing, only message digests are supported for signing.

Reviewer: ```suggestion
-> The current resource is a one-time resource, and destroying this resource will not recover sign, but will only remove the resource information from the tfstate file.

-> 1. This rresource only supports signature operations for asymmetric keys with `key_usage` set to **SIGN_VERIFY**.
  <br/>2. When using SM2 keys for signature, only can be used to sign message digests.
```

Author: ok

## Example Usage

```hcl
variable "key_id" {}
variable "message" {}
variable "signing_algorithm" {}

resource "huaweicloud_kms_sign" "test" {
  key_id            = var.key_id
  message           = var.message
  signing_algorithm = var.signing_algorithm
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.
  Changing this creates a new resource.

* `key_id` - (Required, String, NonUpdatable) Specifies the key ID.
  The valid length is `36` bytes, meeting regular match **^[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}$**.
  For example: **0d0466b0-e727-4d9c-b35d-f84bb474a37f**.

* `message` - (Required, String, NonUpdatable) Specifies the message digest or message to be signed.

>   The message length must be less than `4096` bytes and should be encoded using Base64.

Reviewer: ```suggestion
  The message length must be less than `4,096` bytes and should be encoded using Base64.
```

Author: ok
